client/my-sites/plan-features-2023-grid/index.tsx

```diff
@@ -1180,6 +1148,47 @@ const ConnectedPlanFeatures2023Grid = connect(
 
 		const isJetpackNotAtomic = isJetpack && ! isSiteAT;
 
+		const hasDiscountNotice = () => {
+			const { withDiscount } = ownProps;
+			const bannerContainer = getBannerContainer();
+			if ( ! bannerContainer ) {
+				return false;
+			}
+
+			const activeDiscount = getDiscountByName( withDiscount );
+			if ( ! activeDiscount || hasPlaceholders( planProperties ) || ! canUserPurchasePlan ) {
+				return false;
+			}
+			return true;
+		};
+
+		const higherPlanAvailable = () => {
+			const currentPlan = currentSitePlan?.productSlug;
+			const highestPurchasablePlan = planProperties
+				.filter( ( { planName } ) => planName !== PLAN_ENTERPRISE_GRID_WPCOM )
+				.filter( ( { availableForPurchase } ) => availableForPurchase )
+				.pop();
+			const { planName: highestPurchasablePlanName } = highestPurchasablePlan ?? {};
+			return currentPlan !== highestPurchasablePlanName;
+		};
+
+		const isDiscountNoticeVisible = hasDiscountNotice();
+		const isFreePlanActive = isFreePlan( sitePlan?.product_slug ?? '' );
```

Comment:
the name `isFreePlanActive` is not clear to me. what does it mean (that the site is on a free plan?)

Reply:
Ah yes let me rename that to `isCurrentSiteOnFreeplan`